cpp/arcticdb/processing/test/test_has_valid_type_promotion.cpp

```diff
@@ -121,3 +121,14 @@ TEST(HasValidTypePromotion, FloatFloat) {
     EXPECT_FALSE(has_valid_type_promotion(float64, float32));
 }
 
+TEST(HasValidTypePromotion, EmptyToEverything) {
```

Comment:
An equivalent `EverythingToEmpty` test would also be useful

Reply:
Do we want everything to be convertible to None? I'm not sure how sensible is this.

Reply:
I think so, because if a column has a type already in some other dataframe, but this one is complelely empty, then that's fine, it's just still the same type as it was before

Reply:
Nothing should be promotable to None. `has_valid_common_type` is used if you want to check both ways round.